zone/gm_commands/takeplatinum.cpp

```diff
@@ -0,0 +1,57 @@
+#include "../client.h"
+
+void command_takeplatinum(Client *c, const Seperator *sep)
+{
+	int arguments = sep->argnum;
+	if (!arguments || !sep->IsNumber(1)) { //must be a number
+		c->Message(Chat::Red, "Usage: #Usage: #takeplatinum [Platinum]");
+		return;
+	}
+
+	Client *target = c;
+	if (c->GetTarget() && c->GetTarget()->IsClient()) {
+		target = c->GetTarget()->CastToClient();
+	}
+
+	uint32 platinum = Strings::ToUnsignedInt(sep->arg[1]);
+	if (!platinum) {
+		c->Message(Chat::Red, "Usage: #Usage: #takeplatinum [Platinum]");
+		return;
+	}
+
+	bool success = target->TakePlatinum(
+		platinum,
+		true
+	);
+
+	if (success) {
+		c->Message(
+			Chat::White,
+			fmt::format(
+				"Subtracted {} from {}.",
+				Strings::Money(
+					platinum,
+					0,
+					0,
+					0
+				),
+				c->GetTargetDescription(target)
+			).c_str()
+		);
+	}
+	else {
```

Comment:
Could do a return in `if (success)` instead.

Maybe don’t need to say it’s too much as that’s implied? “You do not have/X does not have Y Platinum.”

Reply:
I think a return is pointless as it is at the end of the code block.

Yes, it was implied (a hint) as that is the main (only?) case I could think that it would fail. If you try to take too much money, it wouldn't tell you. I'll go with the other message you offered.